src/picai_prep/preprocessing.py

```diff
@@ -357,7 +387,11 @@ def preprocess(self):
             self.resample_spacing()
 
         if self.settings.matrix_size is not None or self.settings.physical_size is not None:
-            # perform centre crop
+            # perform centre crop and/or pad
+            self.centre_crop_or_pad()
+
+        if self.settings.max_matrix_size is not None or self.settings.max_physical_size is not None:
```

Comment:
while I know in the post_init neither matrix or max_matrix can be both set, I still think its safer to change this to `elif`